src/main/java/picocli/CommandLine.java

```diff
@@ -6697,7 +6699,7 @@ public void run() {
                 do {
                     if (size.intValue() >= 0) { break; }
                     try { Thread.sleep(25); } catch (InterruptedException ignored) {}
-                } while (System.currentTimeMillis() < now + 2000);
+                } while (System.currentTimeMillis() < now + 2000 && t.isAlive());
```

Comment:
Looks good. Were you able to test that this prevents the 2 second timeout when the pattern cannot match anything (e.g. with the old pattern on a German language environment)?

Reply:
Yes, I manually tested that on my system with the old pattern. The 2 second delay was gone as soon as I introduced the alive test for the thread.

Reply:
Great, thank you!